src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -184,8 +184,18 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         return this.joinConfig?.useKeyDelay ?? 5_000;
     }
 
+    /**
+     * If the server disallows the configured {@link membershipServerSideExpiryTimeout},
+     * this stores a delay that the server does allow.
+     */
+    private membershipServerSideExpiryTimeoutOverride?: number;
+
     private get membershipServerSideExpiryTimeout(): number {
-        return this.joinConfig?.membershipServerSideExpiryTimeout ?? 8_000;
+        return (
+            this.joinConfig?.membershipServerSideExpiryTimeout ??
+            this.membershipServerSideExpiryTimeoutOverride ??
+            8_000
+        );
```

Comment:
```suggestion
        return (
            this.membershipServerSideExpiryTimeoutOverride ??
            this.joinConfig?.membershipServerSideExpiryTimeout ??
            8_000
        );
```

From the wording of the above comment it sounds like these things should go in this order. Have I misunderstood?

Reply:
Whoops, yes! Thanks for catching this!

Reply:
This is now tested: a377b5098